src/pudl_archiver/archivers/validate.py

```diff
@@ -292,3 +304,16 @@ def _validate_xml(buffer: BytesIO) -> bool:
         return False
 
     return True
+
+
+def _validate_csv(buffer: BytesIO) -> bool:
```

Comment:
@zschira This is currently saying, is the csv actually a comma-separated value? But it doesn't give us any information about whether the actual internals of the file are corrupted. Using `pd.read_csv()` on a small slice seems more robust and it sounds like from what you're saying in the issue you agree?

Reply:
Updated to just read the first 100 lines of the CSV. Will run all existing archives against this to make sure skiprows isn't needed, I don't remember this being an issue with our existing CSVs.

Reply:
One system for validating the parameters of a CSV, but not its schema is the [CSV Dialect](https://specs.frictionlessdata.io/csv-dialect/) defined in the Frictionless specs. That seems like the appropriate level of validation to be doing here, with the schema information probably only coming in during the transform step somewhere.

Reply:
This is closer to what I had before (using `csv.Sniffer()`) but it only tells us that this is an actual comma-separated string, not whether it's empty or has indecipherable characters in the header. I think trying to open the first 100 rows of the file seems just as or more useful at this stage, and will catch different types of errors so I'm inclined to stick with it for now. I'm also almost at budget for this step.